### PR TITLE
[docs] Update deployment modes topic and graphics

### LINKED ISSUE #8754: Docs feedback: /fundamentals/architecture/deployment-modes.md
The deployment mode documentation does not explain the new deployment model created in the Helm chart.
The deployment model with a Read Deployment, a Write StatefulSet, and a Backend Deployment.

Can you please provide some further information about this deployment model as it will become the default model for the Loki Helm chart?

### LINKED ISSUE #9765: [Docs] Missing backend component(ssd model)
**Describe the bug**
As of release 2.8.0, a new component for Simple Scalable Mode known as `backend` target was included, however there is no document that refers to this new component.

"Notable changes":

- New backend target:  A new, third target was added to Loki's scalable configuration, which is the default configuration used in the Loki helm chart. This allows Loki to be run as 3 targets (read, write, and backend) and makes the read target stateless and therefore able to be run as a Kubernetes deployment that can be scaled automatically.

**Expected behavior**
A documentation containing this new component, to help new users understand and new implementation options.

**Ref**

- Slack Thread: https://grafana.slack.com/archives/CEPJRLQNL/p1687369520468689
- Components(code): https://github.com/grafana/loki/blob/806674fdaa6d68157019b00534edf3f04f0755d3/pkg/loki/loki.go#L672-L674

### PR DESCRIPTION
**Pull request by JStickler**
**What this PR does / why we need it**:
Revising/Refreshing the Deployment modes as second phase of addressing #9228.

Updates the three graphics:
Monolithic mode diagram simplified to emphasize that this mode is for smaller deployments.
Scalable Monolithic mode diagram now includes backend target.
Microservices mode diagram now includes Compactor, Index Gateway, and Query Scheduler. 

**Which issue(s) this PR fixes**:
Also fixes #9765  (missing backend target) and fixes #8754.

### CODE REVIEW

File: docs/sources/get-started/deployment-modes.md [hunk 1]
```diff
@@ -8,101 +8,81 @@ aliases:
...
+Loki has many component microservices, and is designed to run as a horizontally-scalable distributed system. The Loki binary and Docker image include all of the Loki component microservices, letting you customize how you deploy Loki.
 
-Configuration for each deployed instance of the binary
-further specifies which components it runs.
+You can configure the behavior of the single binary with the `-target` command-line flag to specify which microservices will run on startup. You can further configure each of the components in the `loki.yaml` file.
 
-Loki is designed to easily redeploy a cluster under a different mode
-as your needs change,
-with no configuration changes or minimal configuration changes.
+As Loki stores indexes and chunks separately, you can easily redeploy a cluster under a different mode as your needs change, with minimal or no configuration changes.
```
Comment (slim-bean) [inline docs/sources/get-started/deployment-modes.md:15]:
```suggestion
As Loki decouples the data it stores from the compute which ingests and queries it, you can easily redeploy a cluster under a different mode as your needs change, with minimal or no configuration changes.
```

File: docs/sources/get-started/deployment-modes.md [hunk 2]
```diff
@@ -8,101 +8,81 @@ aliases:
...
 # Loki deployment modes
 
-As an application, 
-Loki is built out of many component microservices,
-and is designed to run as a horizontally-scalable distributed system.
-Loki's unique design compiles the code of the entire distributed system into
-a single binary or Docker image.
-The behavior of that single binary is controlled with
-the `-target` command-line flag and defines one of three modes of operation.
+Loki has many component microservices, and is designed to run as a horizontally-scalable distributed system. The Loki binary and Docker image include all of the Loki component microservices, letting you customize how you deploy Loki.
```
Comment (slim-bean) [inline docs/sources/get-started/deployment-modes.md:11]:
```suggestion
Loki is a distributed system consisting of many microservices.  It also has a unique build model where all of those microservices exist within the same binary.
```

File: docs/sources/get-started/deployment-modes.md [hunk 3]
```diff
@@ -8,101 +8,81 @@ aliases:
...
+As Loki stores indexes and chunks separately, you can easily redeploy a cluster under a different mode as your needs change, with minimal or no configuration changes.
 
-## Monolithic mode
-
-The simplest mode of operation sets `-target=all`.
-This is the default target,
-and it does not need to be specified.
-This is monolithic mode;
-it runs all of Loki’s microservice components inside a single process
-as a single binary or Docker image.
```
Comment (slim-bean) [inline docs/sources/get-started/deployment-modes.md:33]: I'm reluctant to remove this, most of the folks using Loki in the wild run the single binary.

I think we may just want to make clear this is for low volume hobby setups or low volume installations over a fixed number of sources and small number of users.

File: docs/sources/get-started/deployment-modes.md [hunk 4]
```diff
@@ -8,101 +8,81 @@ aliases:
...
 
-Monolithic mode is useful for getting started quickly to experiment with Loki,
-as well as for small read/write volumes of up to approximately 100GB per day.
+Loki’s scalable monolithic deployment mode separates execution paths into read, write, and backend targets. These targets can be scaled independently, letting you customize your Loki deployment to meet your business needs for log ingestion and log query so that your infrastructure costs better match how you use Loki.
 
-Horizontally scale up a monolithic mode deployment to more instances
-by using a shared object store, and by configuring the
-[`ring` section]({{< relref "../configure#common" >}})
-to share state between all instances.
+The scalable monolithic deployment mode can scale to several TBs of logs per day and more.
```
Comment (slim-bean) [inline docs/sources/get-started/deployment-modes.md:23]:
```suggestion
The scalable monolithic deployment mode can scale up to a few TB of logs per day and maybe a little more, however much beyond this and the microservices approach will be advantageous for most users.
```

File: docs/sources/get-started/deployment-modes.md [hunk 5]
```diff
@@ -8,101 +8,81 @@ aliases:
...
 
-* higher availability of write path by providing dedicated nodes
-* separately scalable read path to add/remove query performance on demand
+You can horizontally scale a monolithic mode deployment to more instances by using a shared object store, and by configuring the [`ring` section]({{< relref "../configure#common" >}}) of the `loki.yaml` file to share state between all instances, but the recommendation is to use scalable monolithic mode if you need to scale your deployment.
 
-The simple scalable deployment mode requires a load balancer in front of Loki,
-which directs `/loki/api/v1/push` traffic to the write nodes.
-All other requests go to the read nodes.
-Traffic should be sent in a round robin fashion.
+You can configure high availability by running two Loki instances using `memberlist_config` configuration and a shared object store. You route traffic to all the Loki instances in a round robin fashion.
```
Comment (slim-bean) [inline docs/sources/get-started/deployment-modes.md:53]: note: you also need to enable replication_factor: 3 to get HA with Loki (this is true for SSD as well). Running multiple instances with rep factor of 1 will not give you HA.

File: docs/sources/get-started/deployment-modes.md [hunk 6]
```diff
@@ -8,101 +8,81 @@ aliases:
...
 
-The simple scalable deployment of Loki can scale to
-several TBs of logs per day and more.
+Query parallelization is limited by the number of instances and the setting `max_query_parallelism` which is defined in the `loki.yaml` file.
```
Comment (slim-bean) [inline docs/sources/get-started/deployment-modes.md:55]: I wonder if we should remove this and have a dedicated section on query parallelization/performance.

max_query_parallelism is one thing you care about, but also you will care about max_concurrent, especially in the single binary.

Comment (JStickler) [reply docs/sources/get-started/deployment-modes.md:55]: Maybe in a later PR?  I'd like to get this update in and then we can add additional content later.

**Review by slim-bean — APPROVED** [review]

**Pull request merged**